scripts/util/upload-dependency-script-to-dependency-registry.ts

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+import { Command } from "@commander-js/extra-typings";
+import { ethers } from "hardhat";
+import { DependencyRegistryV0 } from "../contracts";
+import { DependencyRegistryV0__factory } from "../contracts/factories/DependencyRegistryV0__factory";
+import fs from "fs";
+import { ContractTransaction } from "ethers";
+import Graphemer from "graphemer";
+
+const program = new Command()
+  .option(
+    "-d --dependency-registry <dependency-registry>",
+    "Dependency registry address"
+  )
+  .option("-f --file <file>", "Path to file to be uploaded")
+  .option("-t --dependency-type <dependency-type>", "Dependency type")
+  .option(
+    "-o --offset <offset>",
+    "Offset to start at. Usefule for resuming upload after failure"
+  )
+  .option(
+    "-u --update",
+    "Update existing dependency scripts instead of appending"
+  );
+
+program.parse(process.argv);
+
+export const maxContentSize = 24000;
+
+function chunk(s, maxBytes) {
+  let buf = Buffer.from(s);
+  const result = [];
+
+  // Use Graphemer to split on grapheme boundaries
+  const splitter = new Graphemer();
```

Comment:
Note: I wanted this to be set up to handle chunking of any sort of large file so this function is a bit more complex than it needs to be if we expect to only upload base64 encoded, gzipped files which should only really have single byte characters.